app/src/backend/permissions/PermissionService.ts

```diff
@@ -0,0 +1,328 @@
+ * - Role checking (RoleChecker)  
+ * - Resource loading (ResourceLoader)
+ */
+export class PermissionService {
+  
+  // =============================================================================
+  // CORE PERMISSION CHECKING
+  // =============================================================================
+  
+  /**
+   * Main permission check with session
+   */
+  static async checkAccess(
+    session: AppSession | null,
+    context: PermissionContext,
+    options: PermissionOptions = {}
+  ): Promise<ResourceAccess> {
+    if (!session?.user) {
+      logger.warn('Permission check failed: No authenticated session', { context });
+      throw new createHttpError.Unauthorized("Authentication required");
+    }
+    
+    logger.debug('Permission check initiated', {
+      userId: session.user.id,
+      context,
+      options
+    });
+
+    // Default includeResource to true unless explicitly excluded
+    const shouldLoadResource = options.excludeResource ? false : (options.includeResource !== false);
+
+    // System admins bypass all checks
+    if (session.user.role === Roles.Admin) {
+      logger.debug('System admin access granted', { userId: session.user.id });
+      const resource = shouldLoadResource 
+        ? await ResourceLoader.getResource(context) 
+        : undefined;
+      return {
+        hasAccess: true,
+        userRole: UserRole.ORG_ADMIN,
+        organizationId: await PermissionResolver.resolveOrganizationId(context) || 'system',
+        resource
+      };
+    }
+
+    return this.checkUserAccess(session.user.id, context, options);
+  }
+
+  /**
+   * Core user permission check
+   */
+  static async checkUserAccess(
+    userId: string,
+    context: PermissionContext,
+    options: PermissionOptions = {}
+  ): Promise<ResourceAccess> {
+    // Resolve organization context
+    const orgId = await PermissionResolver.resolveOrganizationId(context);
+    if (!orgId) {
+      // If we can't resolve organization context, the resource likely doesn't exist
+      // Return 404 instead of 403 to allow proper API error handling
+      throw createPermissionError(PERMISSION_ERRORS.RESOURCE_NOT_FOUND, 404);
+    }
+
+    // Check organization status if required
+    if (options.requireActive) {
+      await PermissionResolver.ensureOrganizationActive(orgId);
+    }
+
+    // Get user's role in this organization
+    const userRole = await RoleChecker.getUserRoleInOrganization(userId, orgId, context);
+    
+    if (userRole === UserRole.NO_ACCESS) {
+      logger.warn('User has no access to resource', {
+        userId,
+        organizationId: orgId,
+        context
+      });
+      throw createPermissionError(PERMISSION_ERRORS.NO_ACCESS_TO_RESOURCE, 403, {
+        userId,
+        organizationId: orgId,
+        context
+      });
+    }
+    
+    logger.debug('Permission check successful', {
+      userId,
+      userRole,
+      organizationId: orgId
+    });
```

Comment:
Isn't the data object the first argument for the logger functions, before the message?